specutils/manipulation/resample.py

```diff
@@ -0,0 +1,293 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+from scipy.interpolate import CubicSpline
+from astropy.units import Quantity
+from astropy.nddata import StdDevUncertainty, VarianceUncertainty, InverseVariance
+
+from ..spectra import Spectrum1D
+
+__all__ = ['ResampleBase', 'FluxConservingResample']
+
+
+class ResampleBase(ABC):
+    """
+    Base class for resample classes.  The algorithms and needs for difference
+    resamples will vary quite a bit, so this class is relatively sparse.
+
+    Init paramtere here is not yet hooked up to the rest of the code, but to
+    show how we will want to use it in the future.
+    """
+    def __init__(self, bin_edges='zero_fill'):
+        self.bin_edges = bin_edges
+
+    @abstractmethod
+    def __call__(self, orig_spectrum, fin_lamb):
+        """
+        Return the resulting `~specutils.Spectrum1D` of the resampling.
+        """
+        return NotImplemented
+
+    @abstractmethod
+    def resample1D(self, orig_spectrum, fin_lamb):
+        """
+        Workhorse method that will return the resampled Spectrum1D
+        object.
+        """
+        return NotImplemented
+
+    @staticmethod
+    def _calc_bin_edges(x):
+        """
+        Calculate the bin edge values of an input dispersion axis. Input values
+        are assumed to be the center of the bins.
+
+        todo: this should live in the main spectrum object, but we're still
+        figuring out the details to that implementation, so leaving here
+        for now.
+
+        Parameters
+        ----------
+        x : ndarray
+            The input dispersion axis values.
+
+        Returns
+        -------
+        edges : ndarray
+            Calcualated bin edges, including left and right most bin edges.
+        """
+        inside_edges = (x[1:] + x[:-1]) / 2
+        edges = np.insert(inside_edges, 0, 2 * x[0] - inside_edges[0])
+        edges = np.append(edges, 2 * x[-1] - inside_edges[-1])
+
+        return edges
+
+
+class FluxConservingResample(ResampleBase):
+    """
+    This resample algorithim conserves overall flux (as opposed to flux density).
```

Comment:
```suggestion
    This resample algorithim conserves integrated flux (as opposed to flux density).
```

I *think* this is technically more correct - that is, the algorithm conserves the integrated flux in a particular area, not the actual sum-of-fluxes.